src/main/java/com/amplitude/api/AmplitudeClient.java

```diff
@@ -466,6 +467,18 @@ public AmplitudeClient enableNewDeviceIdPerInstall(boolean newDeviceIdPerInstall
      */
     public AmplitudeClient useAdvertisingIdForDeviceId() {
         this.useAdvertisingIdForDeviceId = true;
+        this.useAppSetIdForDeviceId = false;
+        return this;
+    }
+
+    /**
+     * Use Android app set id as the user's device ID.
+     *
+     * @return the AmplitudeClient
+     */
+    public AmplitudeClient useAppSetIdForDeviceId() {
+        this.useAppSetIdForDeviceId = true;
```

Comment:
Do you think this should be an enum Qingzhuo? We might have more options in the future

Reply:
@dantetam Could you elaborate a bit more on the enum idea? Do you mean like a separate variable like `ForDeviceId` to hold a enum for the options?

Reply:
Yes, like that. Having multiple booleans like `useAppSetIdForDeviceId` and `useAdvertisingIdForDeviceId` and so on will get confusing in my opinion

Reply:
I see, let me replace these two with enum

Reply:
just replaced to use enum @dantetam